src/extension/tags.c

```diff
@@ -856,6 +868,81 @@ bool is_user_id_sensitive(zend_string *user_id)
         match_regex(_uuid_zstr, user_id) || match_regex(_id_zstr, user_id));
 }
 
+static PHP_FUNCTION(datadog_appsec_track_user_signup_event)
+{
+    UNUSED(return_value);
+    if (DDAPPSEC_G(enabled) != ENABLED) {
+        mlog(dd_log_debug, "Trying to access to track_user_signup_event "
+                           "function while appsec is disabled");
+        return;
+    }
+
+    zend_string *user_id = NULL;
+    HashTable *metadata = NULL;
+    zend_bool automated = false; // Don't document. Only internal usage
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "S|hb", &user_id, &metadata,
+            &automated) == FAILURE) {
+        mlog(dd_log_warning, "Unexpected parameter combination, expected "
+                             "(user_id, metadata)");
+        return;
+    }
+    if (automated) {
+        if (automated_user_events_tracking == NOT_ENABLED) {
+            return;
+        }
+
+        if (automated_user_events_tracking == SAFE) {
+            if (user_id != NULL && is_user_id_sensitive(user_id)) {
+                zend_string_release(user_id);
```

Comment:
I didn't think about this before, but if the user was passing its own string, would this invalidate it? Same with the metadata, would the user-space array be empty?

Reply:
Let me check it and come back to you

Reply:
It happens on the metadata var but not on the user id. I will change it